src/braket/default_simulator/simulator.py

```diff
@@ -407,7 +407,21 @@ def _formatted_measurements(
         ]
         #  Gets the subset of measurements from the full measurements
         if measured_qubits is not None and measured_qubits != []:
-            measurements = np.array(measurements)[:, measured_qubits].tolist()
+            if any(qubit in range(simulation.qubit_count) for qubit in measured_qubits):
+                measured_qubits_in_circuit = [
```

Comment:
this part does two list-comprehensions to find qubits greater or less than simulation.qubit_count. 

This could be more efficient using numpy with boolean indexing, which is vectorized and very fast. Additionally, the condition is only written once below:
```
measured_qubits = np.array(measured_qubits)
in_circuit_mask = measured_qubits < simulation.qubit_count

measured_qubits_in_circuit = measured_qubits[in_circuit_mask]

measured_qubits_not_in_circuit = measured_qubits[~in_circuit_mask]
```

Reply:
Updated this to use boolean indexing.